solidity/test/maintainer/MaintainerProxy.test.ts

```diff
@@ -0,0 +1,3434 @@
+              ...data,
+              wallet: {
+                ...data.wallet,
+                state: walletState.MovingFunds,
+              },
+            })
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should succeed", async () => {
+            await expect(outcome).to.not.be.reverted
+          })
+
+          it("should refund ETH", async () => {
+            const resolvedOutcome = await outcome
+
+            const postThirdPartyBalance = await provider.getBalance(
+              authorizedMaintainer.address
+            )
+
+            const diff = postThirdPartyBalance.sub(
+              resolvedOutcome.initThirdPartyBalance
+            )
+            expect(diff).to.be.gt(0)
+            expect(diff).to.be.lt(
+              ethers.utils.parseUnits("1000000", "gwei") // 0,001 ETH
+            )
+          })
+        })
+      })
+    })
+  })
+
+  describe("notifyWalletClosingPeriodElapsed", () => {
+    const walletDraft = {
+      ecdsaWalletID: ecdsaWalletTestData.walletID,
+      mainUtxoHash: ethers.constants.HashZero,
+      pendingRedemptionsValue: 0,
+      createdAt: 0,
+      movingFundsRequestedAt: 0,
+      closingStartedAt: 0,
+      pendingMovedFundsSweepRequestsCount: 0,
+      state: walletState.Unknown,
+      movingFundsTargetWalletsCommitmentHash: ethers.constants.HashZero,
+    }
+
+    context("when the wallet is in the Closing state", () => {
+      before(async () => {
+        await createSnapshot()
+
+        await bridge.setWallet(ecdsaWalletTestData.pubKeyHash160, {
+          ...walletDraft,
+          state: walletState.Live,
+        })
+
+        // Switches the wallet to Closing state because the wallet has
+        // no main UTXO set.
+        await bridge
+          .connect(walletRegistry.wallet)
+          .__ecdsaWalletHeartbeatFailedCallback(
+            ecdsaWalletTestData.walletID,
+            ecdsaWalletTestData.publicKeyX,
+            ecdsaWalletTestData.publicKeyY
+          )
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      context("when closing period has elapsed", () => {
+        let tx: ContractTransaction
+        let initThirdPartyBalance: BigNumber
+
+        before(async () => {
+          await createSnapshot()
+
+          await increaseTime(
+            (
+              await bridge.walletParameters()
+            ).walletClosingPeriod
+          )
+
+          initThirdPartyBalance = await provider.getBalance(
+            authorizedMaintainer.address
+          )
+          tx = await maintainerProxy
+            .connect(authorizedMaintainer)
+            .notifyWalletClosingPeriodElapsed(ecdsaWalletTestData.pubKeyHash160)
+        })
+
+        after(async () => {
+          await walletRegistry.closeWallet.reset()
+
+          await restoreSnapshot()
+        })
+
+        it("should successfully submit sweep proof", async () => {
```

Comment:
Errr... no sweep proofs here! Let's switch to events.

Reply:
https://github.com/keep-network/tbtc-v2/pull/209/commits/0c1261b569df4efca23555f3408add0172c9c31f